core/src/main/java/org/infinispan/remoting/transport/jgroups/SecurityActions.java

```diff
@@ -0,0 +1,63 @@
+package org.infinispan.remoting.transport.jgroups;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+
+/**
+ * Privileged actions for the package
+ *
+ * Do not move. Do not change class and method visibility to avoid being called from other
+ * {@link java.security.CodeSource}s, thus granting privilege escalation to external code.
+ *
+ * @author Scott.Stark@jboss.org
+ * @since 4.2
+ */
+final class SecurityActions {
+
+   interface SysProps {
+
+      SysProps NON_PRIVILEGED = new SysProps() {
+         @Override
+         public boolean getBooleanProperty(String name) {
+            return Boolean.getBoolean(name);
+         }
+
+         @Override
+         public int getIntProperty(String name, int defaultValue) {
+            return Integer.getInteger(name, defaultValue);
+         }
+      };
+
+      SysProps PRIVILEGED = new SysProps() {
+         @Override
+         public boolean getBooleanProperty(final String name) {
+            PrivilegedAction<Boolean> action = () -> Boolean.getBoolean(name);
+            return AccessController.doPrivileged(action);
+         }
+
+         @Override
+         public int getIntProperty(final String name, int defaultValue) {
+            PrivilegedAction<Integer> action = () -> Integer.getInteger(name, defaultValue);
+            return AccessController.doPrivileged(action);
+         }
+      };
+
+      boolean getBooleanProperty(String name);
+
+      int getIntProperty(String name, int defaultValue);
+   }
+
+   static boolean getBooleanProperty(String name) {
+      if (System.getSecurityManager() == null)
```

Comment:
I commented above. @danberindei should be using a SecurityAction approach instead.